types/src/Mooclet/MoocletTSConfigurablePolicyParametersDTO.ts

```diff
@@ -1,39 +1,78 @@
-import { IsNumber, IsString, ValidateNested, IsOptional, IsObject, IsDefined, IsNotEmpty } from 'class-validator';
-import { Type } from 'class-transformer';
+import {
+  IsNumber,
+  IsString,
+  ValidateNested,
+  IsOptional,
+  registerDecorator,
+  IsDefined,
+  IsNotEmpty,
+  ValidationOptions,
+  validate as CVValidate,
+  validateSync as CVValidateSync,
+} from 'class-validator';
+import { plainToClass, Type } from 'class-transformer';
 import { MoocletPolicyParametersDTO } from './MoocletPolicyParametersDTO';
 
 export class Prior {
   @IsDefined()
   @IsNumber()
-  @Type(() => Number)
-  failure = 1;
+  failure: number;
 
   @IsDefined()
   @IsNumber()
-  @Type(() => Number)
-  success = 1;
+  success: number;
 }
 
 export class CurrentPosteriors {
+  @IsDefined()
   @IsNumber()
-  @Type(() => Number)
   failures: number;
 
+  @IsDefined()
   @IsNumber()
-  @Type(() => Number)
   successes: number;
 }
 
+const IsCurrentPosteriorsRecord = (validationOptions?: ValidationOptions) => {
+  return function (object: unknown, propertyName: string) {
+    registerDecorator({
+      name: 'IsCurrentPosteriorsRecord',
+      target: object.constructor,
+      propertyName: propertyName,
+      constraints: [],
+      options: {
+        message: ({ value }) => {
+          const errors = Object.values(value).map((val) => {
+            const instance = plainToClass(CurrentPosteriors, val);
+            return CVValidateSync(instance);
+          });
+          // Flat the array of arrays
+          return errors.reduce((acc, val) => acc.concat(val), [])[0].toString();
+        },
+        ...validationOptions,
+      },
+      validator: {
+        async validate(value: unknown) {
+          return Promise.all(
+            Object.values(value).map(async (val) => {
+              const instance = plainToClass(CurrentPosteriors, val);
+              return await CVValidate(instance).then((errors) => errors.length === 0);
+            })
+          ).then((values) => values.every((val) => val));
+        },
+      },
+    });
+  };
+};
+
 export class MoocletTSConfigurablePolicyParametersDTO extends MoocletPolicyParametersDTO {
   @IsDefined()
   @ValidateNested()
   @Type(() => Prior)
-  prior: Prior = new Prior();
+  prior: Prior;
```

Comment:
Without the default values for prior and it's properties, the frontend can't create a default object, the priors as `"prior": { "success": 1, "failure": 1 }` should show up here in the editor.

<img width="1008" alt="image" src="https://github.com/user-attachments/assets/122a3cdf-49f7-426e-afcf-ec5820cc770b" />

Reply:
The validation isn't quite working for me on the frontend when I fill in prior manually either, it allows me to hit "next" with an object like below. It does catch it on submit and give correct message, however the other validations will raise an error immediately as the user types.

{
  "prior": {
    "success": "d",
    "failure": 2
  },

Reply:
We should not add the default values in the class constructor because class transform uses this class constructor to create the definition and assign the object (to be tested) to the constructed object. If we are adding the default values, the constructor definition also contains the value which we don't want.
For testing, add the default value for `Prior` in the types
Remove the `"prior": { "success": 1, "failure": 1}`. 
It will not throw an error.

I can add the appropriate config for the frontend side for validation to work correctly.